packages/next/taskfile-swc.js

```diff
@@ -118,7 +118,25 @@ module.exports = function (task) {
         if (interopClientDefaultExport) {
           output.code += `
 if (typeof exports.default === 'function' || (typeof exports.default === 'object' && exports.default !== null)) {
-  Object.assign(exports.default, exports);
+  for (var exp in exports) {
+    if (Object.prototype.hasOwnProperty.call(exports, exp)) {
+      Object.defineProperty(exports.default, exp, {
+        enumerable: true,
+        configurable: true,
+        get: (function (exp) {
+          return function() {
+            return exports[exp];
+          }
+        })(exp),
+        set: (function (exp) {
+          return function(newValue) {
+            exports[exp] = newValue;
+          }
+        })(exp)
+      });
+    }
+  }
+  Object.defineProperty(exports.default, '__esModule', { value: true });
```

Comment:
IMHO:

```js
module.exports = exports = Object.assign(exports.default, exports);
```

This should be able to retain reference between `module.exports`, `exports` and `exports.default`?